RealmSwift/SwiftUI.swift

```diff
@@ -232,16 +232,7 @@ private final class ObservableStoragePublisher<ObjectType>: Publisher where Obje
 
 @available(iOS 13.0, macOS 10.15, tvOS 13.0, watchOS 6.0, *)
 private class ObservableStorage<ObservedType>: ObservableObject where ObservedType: RealmSubscribable & ThreadConfined & Equatable {
-    @Published var value: ObservedType {
-        willSet {
-            if newValue != value {
-                objectWillChange.subscribers.forEach {
-                    $0.receive(subscription: ObservationSubscription(token: newValue._observe(keyPaths, $0)))
-                }
-                objectWillChange.send()
-            }
-        }
-    }
+    @Published var value: ObservedType
```

Comment:
Given that `ObservableStorage` is not used exclusively by `@ObservedResults` and `@ObservedSectionedResults`, wouldn't this change affect also `@StateRealmObject` and  `@ObservableRealmObject` in cases when the value is updated and we need to update the subscription. For example
``` swift
struct DetailTicketView: View {
    @StateRealmObject var ticket: Ticket = Ticket()
    @Environment(\.realm) var realm

    var body: some View {
        VStack {
            TextField("Title", text: $ticket.title)
                .font(.headline)
        }
        .onAppear {
            ticket = realm.objects(Ticket.self).first!
        }
    }
}
``` I don’t know if this is a common use case, but may be affecting users that use `@StateRealmObject` as a draft object. I also tested things and the behaviour changes with your solution.  I’ve been investigating this and changing the order of calling `objectWillChange` in the willSet removes the warning
``` swift @Published var value: ObservedType {
        willSet {
            if newValue != value {
                objectWillChange.send()
                objectWillChange.subscribers.forEach {
                    $0.receive(subscription: ObservationSubscription(token: newValue._observe(keyPaths, $0)))
                }
            }
        }
}  ``` Which makes sense here because calling `subscriber.receive()` tells the subscriber that a publisher is ready to receive further requests, and then we tell it to update given subscription by calling `subscriber.receive(subscription)`